### DISCUSSION TITLE
Integrating your editor into an electronjs app

### DISCUSSION THREAD

**Discussion by AndryGabry01** [jgraph/drawio#3662] [General]
is it possible to integrate the draw.io editor into your electron js application and interact with the editor itself?

also is there an email or contact for commercial requests?

Essentially I would like to create an application that implements your editor as a base, and then add some of my own functionality to it (which I would prefer to talk about in private).

Thank you for your attention and for your future response.

**Comment by davidjgraph** [comment]
> is it possible to integrate the draw.io editor into your electron js application and interact with the editor itself?

I don't understand this. The draw.io editor integrated into electron is what the desktop is. Electron is not our application.

> also is there an email or contact for commercial requests?

No, this is an open source project.

**Reply by AndryGabry01** [reply]
My idea is to create my own application that takes advantage of the draw.io editor.
What I would need is to integrate the map creation functionality into my own app.
To do this, assuming I choose to use electron, is there an sdk or documentation that shows how to implement this?
Like a nodejs module or something?

Implementation aside, legally speaking, does the draw.io license allow me to use their code in my app?

Thank you very much for your attention and helpfulness.

**Reply by davidjgraph** [reply]
> To do this, assuming I choose to use electron, is there an sdk or documentation that shows how to implement this?

We don't have any documentation on integration. Plenty of developers have implemented integrations, but it's a task for someone with a lot of development experience.

> Implementation aside, legally speaking, does the draw.io license allow me to use their code in my app?

The Apache 2.0 license tells you what you can and can't do with it. Unfortunately, "can I use it" can only ever be answered with "it depends, read the license". If the license could be written in a shorter form that explains everything, it would be shorter. It anyone gives you a different answer, either they a) fully understand everything about your software, it's implementation and dependencies or b) don't understand software licensing.

**Reply by AndryGabry01** [reply]
ah okey, I saw that draw.io-desktop exists.
I'll try to take a cue from that.
In any case, you shortened my time for research and etc and for that I am very grateful.

Thanks again for your helpfulness 😊.
